src/Components/Common/AzureComponent.cs

```diff
@@ -53,7 +53,10 @@ internal void AddClient(
         var configSection = builder.Configuration.GetSection(configurationSectionName);
 
         var settings = new TSettings();
+        // Bind both top-level and named configuration sections to the settings object
+        // to allow connection-specific settings.
         BindSettingsToConfiguration(settings, configSection);
+        BindSettingsToConfiguration(settings, configSection.GetSection(connectionName));
```

Comment:
Why are we only doing this for Azure client libraries? Shouldn't we be doing it for all, so they are consistent?

Reply:
Capturing offline discussion, we'll do this for all integrations but prioritizing Azure components and dependencies for Functions in this PR.